server/engine/engine.go

```diff
@@ -290,6 +291,41 @@ func (engine *Engine) waitForStepRestart(execution *model.Execution, waitGroup *
 func (engine *Engine) runStep(step model.Step, execution *model.Execution, waitGroup *sync.WaitGroup) {
 	defer waitGroup.Done()
 
+	// Deployment pre-check
+	if step.Name == model.WHAT_IF_STEP_NAME {
+		start := time.Now()
+		execution.CorrelationID = "N/A"
+		whatIf, err := azure.StartWhatIf(engine.context, engine.deploymentsClient, step.Name, step.Template, step.Parameters)
+		if err != nil {
+			log.Printf("Failed to start [%s]: %v", step.Name, err)
```

Comment:
Not certain if the err format back from StartWhatIf is the same as from an actual deployment, should check that.